### PR TITLE
Added async-execute as default execution mode in process job control options

### LINKED ISSUE #248: [Feature] Default execution mode reported in process job control options
Implement [/req/ogc-process-description/default-mode](https://github.com/opengeospatial/ogcapi-processes/blob/01f15b0ce0016e162c1006b8c529dfd19e43c4dc/core/requirements/ogc-process-description/REQ_default_mode.adoc)

> The first execution mode specified in the array of job control options in the process description SHALL be the default execution mode.

From https://github.com/opengeospatial/ogcapi-processes/pull/182

All processes should indicate `async-execute` in Weaver's case since this is the implemented method.

### PR DESCRIPTION
**Pull request by trapsidanadir**
Default execution mode reported in process job control options (fixes #248)

### CODE REVIEW

**Comment by fmigneault** [comment]
@trapsidanadir  I converted the PR to draft. Convert it back and tag my once ready.

**Comment by trapsidanadir** [comment]
@fmigneault it's ready for review

File: tests/wps_restapi/test_processes.py [hunk 1]
```diff
@@ -160,6 +160,7 @@ def test_get_processes(self):
             assert "version" in process and isinstance(process["version"], str)
             assert "keywords" in process and isinstance(process["keywords"], list)
             assert "metadata" in process and isinstance(process["metadata"], list)
+            assert len(process["jobControlOptions"]) > 0
```
Comment (fmigneault) [inline tests/wps_restapi/test_processes.py:163]: validate against explicit 1, and that the returned value is the expected default execution mode

**Review by fmigneault — CHANGES_REQUESTED** [review]
Please update the CHANGES file to provide the details about the fix. 
It should be put under the `Unreleased->Fixes` part.

**Comment by fmigneault** [comment]
@trapsidanadir 
Just thought about something else I forgot to add to the review. 
We should have another test that validates the other user case where the value is provided in the request body. 
2 requests should be done, 1 that explicitly provides the `async-execute` and another for the `sync-execute`.
In both cases, the value that was explicitly provided should be the one that is returned in the list.

File: tests/wps_restapi/test_processes.py (hunk 1 above)
Comment (trapsidanadir) [reply tests/wps_restapi/test_processes.py]: Changed, see last commit

File: tests/wps_restapi/test_processes.py [hunk 2]
```diff
@@ -160,6 +160,7 @@ def test_get_processes(self):
             assert "version" in process and isinstance(process["version"], str)
             assert "keywords" in process and isinstance(process["keywords"], list)
             assert "metadata" in process and isinstance(process["metadata"], list)
+            assert len(process["jobControlOptions"]) == 1 and 'async-execute' in process["jobControlOptions"]
```
Comment (fmigneault) [inline tests/wps_restapi/test_processes.py:163]: please use the variable instead of the string

**Review by fmigneault — CHANGES_REQUESTED** [review]
Please check the test output. It reports a linting issue. 
Can be merged once everything passes and my other comment is resolved.

File: tests/wps_restapi/test_processes.py (hunk 2 above)
Comment (trapsidanadir) [reply tests/wps_restapi/test_processes.py]: Solved, see last commit

**Review by fmigneault — APPROVED** [review]

**Pull request merged**